packages/client/src/api.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+import { resolveSelect } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { store } from './store';
+import type { Ability, AbilityInput, AbilityOutput } from './types';
+
+/**
+ * List all available abilities.
+ *
+ * @return Promise resolving to array of abilities.
+ */
+export async function listAbilities(): Promise< Ability[] > {
+	return await resolveSelect( store ).getAbilities();
+}
+
+/**
+ * Get a specific ability by name.
+ *
+ * @param name The ability name.
+ * @return Promise resolving to the ability or null if not found.
+ */
+export async function getAbility( name: string ): Promise< Ability | null > {
+	return await resolveSelect( store ).getAbility( name );
+}
+
+/**
+ * Execute an ability.
+ *
+ * Uses apiFetch since this is a custom action endpoint, not a standard REST resource.
+ * The method (GET or POST) is determined by the ability's type metadata.
+ *
+ * @param name  The ability name.
+ * @param input Optional input parameters for the ability.
+ * @return Promise resolving to the ability execution result.
+ * @throws Error if the ability is not found.
+ */
+export async function executeAbility(
+	name: string,
+	input: AbilityInput = {}
+): Promise< AbilityOutput > {
+	const ability = await getAbility( name );
+	if ( ! ability ) {
+		throw new Error( `Ability not found: ${ name }` );
+	}
+
+	const isResource = ability.meta?.type === 'resource';
+	const method = isResource ? 'GET' : 'POST';
+
+	let path = `/wp/v2/abilities/${ name }/run`;
+	const options: {
+		method: string;
+		data?: { input: AbilityInput };
+	} = {
+		method,
+	};
+
+	if ( method === 'GET' && Object.keys( input ).length > 0 ) {
+		// e.g., input[format]=iso&input[timezone]=UTC
+		const params = new URLSearchParams();
+		Object.entries( input ).forEach( ( [ key, value ] ) => {
+			params.append( `input[${ key }]`, String( value ) );
+		} );
+		path = `${ path }?${ params.toString() }`;
+	} else if ( method === 'POST' ) {
+		options.data = { input };
+	}
+
+	try {
+		return await apiFetch< AbilityOutput >( {
```

Comment:
@mamaduka or @jsnajdr, do we have actions in store that are thunks and alone don't update the state? I'm mostly interested in understanding whether the proposed implementation, expressed also as an action, would bring any benefits or only confusion.

Reply:
As @gziolo noted, this store is also used for client-only abilities and in the future, potentially hybrid abilities. 

I think a store for this makes sense (keeping in mind client-only abilities from #69) -- but I'm also open to other suggestions.